rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -714,80 +730,89 @@ public ResponseEntity linkToProjects(
                             // TODO: Add example for MAP value
                     }
             )
-            @RequestBody List<String> projectIdsInRequestBody
+            @RequestBody List<String> projectIdsInRequestBody,
+            @RequestParam(value = "comment", required = false) String comment
     ) throws URISyntaxException, TException {
 
     User sw360User = restControllerHelper.getSw360UserFromAuthentication();
     Project sourceProj = projectService.getProjectForUserById(id, sw360User);
     Map<String, String> responseMap = new HashMap<>();
     HttpStatus status = null;
-    Set<String> alreadyLinkedIds= new HashSet<>();
+    Set<String> alreadyLinkedIds = new HashSet<>();
     Set<String> idsSentToModerator = new HashSet<>();
     Set<String> idsWithCyclicPath = new HashSet<>();
     Set<String> linkedProjectIds = new HashSet<>();
     int count = 0;
 
     try {
 
-        for(String projId: projectIdsInRequestBody) {
+        for(String projId : projectIdsInRequestBody){
             Project proj = projectService.getProjectForUserById(projId, sw360User);
-            Map<String, ProjectProjectRelationship> linkedProject = Optional.ofNullable(proj.getLinkedProjects()).orElse(new HashMap<>());
+            Map<String, ProjectProjectRelationship> linkedProject = Optional.ofNullable(proj.getLinkedProjects())
+                    .orElse(new HashMap<>());
 
             if (linkedProject.keySet().contains(id)) {
                 alreadyLinkedIds.add(projId);
                 continue;
             }
 
-            linkedProject.put(id, new ProjectProjectRelationship(ProjectRelationship.CONTAINED).setEnableSvm(sourceProj.isEnableSvm()));
-            proj.setLinkedProjects(linkedProject);
-            String cyclicLinkedProjectPath = projectService.getCyclicLinkedProjectPath(proj, sw360User);
-            if (!CommonUtils.isNullEmptyOrWhitespace(cyclicLinkedProjectPath)) {
-                idsWithCyclicPath.add(cyclicLinkedProjectPath);
-                continue;
-            }
+            sw360User.setCommentMadeDuringModerationRequest(comment);
+            if (!restControllerHelper.isWriteActionAllowed(proj, sw360User) && comment == null) {
+                return new ResponseEntity(RESPONSE_BODY_FOR_MODERATION_REQUEST_WITH_COMMIT, HttpStatus.ACCEPTED);
```

Comment:
In case of missing request parameter, shouldn't the response status be `BAD_REQUEST`?